database-commons/src/main/java/io/cdap/plugin/db/batch/source/AbstractDBSource.java

```diff
@@ -471,6 +475,11 @@ public boolean canConnect() {
         !containsMacro(ConnectionConfig.USER) && !containsMacro(ConnectionConfig.PASSWORD) &&
         !containsMacro(DBSourceConfig.DATABASE) && !containsMacro(DBSourceConfig.IMPORT_QUERY);
     }
+
+    @Override
+    public Integer getFetchSize() {
+      return 0;
```

Comment:
Why don't we add proper fetchSize here?

Reply:
It was the default implementation but as discussed offline, we can make this abstract when adding the fetchSize config in the remaining db-plugins.